### ISSUE TITLE
Add Events and Event changes from SpigotAPI

### ISSUE THREAD

**Issue by ZephireNZ** [GlowstoneMC/Glowkit-Legacy#18]
Spigot has added some useful events and misc changes that we may as well merge into Glowkit. These are:
- [PlayerItemDamageEvent](https://github.com/SpigotMC/Spigot-API/blob/master/src/main/java/org/bukkit/event/player/PlayerItemDamageEvent.java)
- [EntitySpawnEvent and SpawnerSpawnEvent (and edits of others)](https://github.com/SpigotMC/Spigot-API/commit/db3880b8008068840029097732749de28d2eb337)
- [getClickedInventory in InventoryClickEvent](https://github.com/SpigotMC/Spigot-API/commit/299f30d21c7f8c61472f11eb384e013c8c4dba23)
- [EntityMountEvent and EntityDismountEvent](https://github.com/SpigotMC/Spigot-API/commit/738ab38374d7cf01c2ca4e649f8a142c89a341bb)
- [PlayerSpawnLocationEvent](https://github.com/SpigotMC/Spigot-API/commit/72bcf6063620f8af50af33e200886a865538c15b)

A full changelist for SpigotAPI can be found here (first spigot change is "POM Changes": 
https://github.com/SpigotMC/Spigot-API/commits/master

**Comment by SpaceManiac** [comment]
I am not averse to porting some of the events or new methods over where they make sense on an individual basis, but the `.spigot()` construct doesn't seem appropriate for Glowkit/Glowstone use. If there's something specific you're after (which makes sense with Glowstone's current state), feel free to PR it.

**Comment by ZephireNZ** [comment]
True. Just thought I'd create the issue, and leave it up to someone if they wanted to pull in any of the events.

**Comment by mastercoms** [comment]
I am going to make a new pull request with only things Spigot has added that do not adversely change Bukkit behavior. Then, we can discuss the changes Spigot has made by deleting Bukkit code.
